### ISSUE TITLE
summing a list of qobj in qutip 5.0

### ISSUE THREAD

**Issue by shohamjac** [qutip/qutip#1841]
### Bug Description

Summing a list of qobj does not work anymore since the change in the qobj class.

### Code to Reproduce the Bug

```shell
from qutip import *
qobj_list = [fock(3,i) for i in range(3)]
sum(qobj_list)
```

### Code Output

```shell
Traceback (most recent call last):
  File "C:\Users\shoha\miniconda3\envs\qutip_fork\lib\site-packages\IPython\core\interactiveshell.py", line 3361, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-6-27c43f8aaf45>", line 1, in <cell line: 1>
    sum(qobj_list)
  File "C:\Users\shoha\miniconda3\envs\qutip_fork\lib\site-packages\qutip-5.0.0.dev0+8187a26-py3.9-win-amd64.egg\qutip\core\qobj.py", line 443, in __radd__
    return self.__add__(other)
  File "C:\Users\shoha\miniconda3\envs\qutip_fork\lib\site-packages\qutip-5.0.0.dev0+8187a26-py3.9-win-amd64.egg\qutip\core\qobj.py", line 153, in out
    raise ValueError(msg)
ValueError: incompatible dimensions [[3], [1]] and [[1], [1]]
```

### Expected Behaviour

Summing should be equal to (qobj_list[0]+qobj_list[1]+....+qobj_list[N-1])

### Your Environment

```shell
QuTiP Version:      5.0.0.dev0+8187a26
Numpy Version:      1.16.6
Scipy Version:      1.7.3
Cython Version:     0.29.25
Matplotlib Version: 3.4.3
Python Version:     3.9.7
Number of CPUs:     4
BLAS Info:          INTEL MKL
INTEL MKL Ext:      False
Platform Info:      Windows (AMD64)
```

### Additional Context

If I understand correctly, this happens because sum starts with 0 and then every object in the list, as written here:
http://www.marinamele.com/2014/04/modifying-add-method-of-python-class.html

**Comment by Ericgig** [comment]
In python `sum` start at `0`. 
So it's doing `(0 + qobj_list[0] + ...)`
And `int + Qobj` is only defined for square `Qobj`.
`sum(qobj_list[1:], start=qobj_list[0])` would work.

**Comment by hodgestar** [comment]
Fixed in #1934.
